framework/src/interfaces/Coupleable.C

```diff
@@ -2140,10 +2140,23 @@ Coupleable::coupledIndices(const std::string & var_name) const
   return coupledVectorHelper<unsigned int>(var_name, func);
 }
 
+VariableName
+Coupleable::coupledName(const std::string & var_name, unsigned int comp) const
+{
+  if (getVar(var_name, comp))
+    return getVar(var_name, comp)->name();
+  // For constants, we'll just return the constant as the name
+  else if (MooseUtils::parsesToReal(var_name))
+    return var_name;
+  else
+    mooseError(
+        _c_name, ": Variable '", var_name, "' does not exist, yet its coupled name is requested");
+}
```

Comment:
I wonder if we should make `coupledName`  error out if a constant was coupled. I.e. remove line 2149 and 2150. This would force users to use `isCoupledConstant`. 

We could potentially add `coupledConstant` to recover the coupled constant value.

I just don;t see a usecase for working with a string representation of a coupled constant number literal.

Reply:
I can see new users wondering why material properties like `dF/d12.721` are created.

Reply:
You could keep the lines but make 2150 emit a different `mooseError`. E.g. "The name of a coupled variable was requested, but the user provided a coupled constant value instead."

Reply:
This is more general than material properties though. coupledName can be used in unrelated code. Do we want to ban constants as soon as the routine relies on the name (for an error message for example?) of the variable?

Reply:
do dF/d12.711 appear from the getMaterialDerivative calls? Do they get created by default instead of a 0 matprop?

Reply:
you would have to suppress creation with a call to isCoupledConstant. Erroring would force the dev to do that. Also, are we using variable names in error messages? We could have a separate call for that which returns the constant string. In fact we already do, it is: `getParam<std::vector<VariableName>>("var_param_name")[component]`

Reply:
There are 100 or so occurences of getVar->name(), which because they seg fault with constants have been changed to coupledName().

If we made them error instead, it's  definitely better than a seg fault, but it's lost capability. We could no longer use constants there for a while until the devs fix them all. 

Which is unlikely to happen, because the devs did not fix it when it was a seg fault, so when it s a benign error..

imo, the next improvement should not lie with coupledName() but with declareMatPropDerivative, that would avoid declaring a derivative if the variable is a constant.

Reply:
How is this lost capable of it never worked?!

Reply:
It works now :)

Reply:
> Erroring would force the dev to do that.

The seg fault did not stop the devs from doing it.

I'll summarize my position on this:
- I want coupledName to be used because the getVar path is not working
- I dont want the spread of coupledName to limit our ability to pass constants as variables

Reply:
@dschwen I caved in
Can I please get an approval :)

Reply:
I agree with both of your bullets, but the spread of coupledName is not limiting our ability to pass constants as variables! In the cases where we used getVar instead of coupledName we didn't ever have that capability in the first place! We'd get a segfault instead of an error. Prior to this PR to have this capability the developer would have to explicitly check for a nullptr, after this PR the developer will have to use `isCoupledConstant`.

Overall the PR is an improvement because you are
1. turning a segfault into a proper mooseError
2. more readable functions will be used that convey intent